tslib/src/caching/redis-service.ts

```diff
@@ -41,26 +72,77 @@ export class RedisService implements IRedisService {
       }
     }
     this._redis = new Redis(options)
+    this._logger = new Logger.ConsoleLogger()
   }
 
   async get (key: string): Promise<any | null> {
     try {
-      const result = await this._redis.get(key)
-      return result
+      const cacheEntry = await this._redis.get(key)
+      if (cacheEntry === null) return Promise.resolve(null)
+      const cacheItem: RedisCacheItem = JSON.parse(cacheEntry)
+      switch (cacheItem._meta.type) {
+        case 'string':
+          return cacheItem.value
+        case 'number':
+          return Number(cacheItem.value)
+        case 'object':
+          const hydratedObject = JSON.parse(cacheItem.value)
```

Comment:
hmm, that could suppress an issue with the set methods, as it would suggest that it wasn't typed properly when persisted.  maybe we should discuss and explore further